base/task.jl

```diff
@@ -365,6 +365,142 @@ function wait(t::Task)
     nothing
 end
 
+# Wait multiple tasks
+
+"""
+    waitany(tasks; throw=false) -> (done_tasks, remaining_tasks)
+
+Wait until at least one of the given tasks have been completed.
+
+If `throw` is `true`, throw `CompositeException` when one of the
+completed tasks completes with an exception.
+
+The return value consists of two task vectors. The first one consists of
+completed tasks, and the other consists of uncompleted tasks.
+"""
+waitany(tasks; throw=false) = _wait_multiple(tasks, throw)
+
+"""
+    waitall(tasks; failfast=false, throw=false) -> (done_tasks, remaining_tasks)
```

Comment:
I wonder if we should switch these defaults? `throw` is the default behavior for `wait`, so I think that would be more consistent with it. It seems like `failfast` is often also a commonly desired behavior, though I am less certain of whether that is a good choice of default. (we can also merge and then change our minds about the default anytime before the next release)
```suggestion
    waitall(tasks; failfast=true, throw=true) -> (done_tasks, remaining_tasks)
```

Reply:
I agree with you about `throw`'s default value.  The consistency with the existing APIs is critical.

Speaking of `failfast`, I don't know a strong preference for it.  I just selected `false` based on my feelings.  Simple `waitall(tasks)` seems to be waiting for all the tasks to be completed for me.  But, the default value should be selected for usefulness.  If it seems that `failfast=true` would occur more frequently, I'd like to change the default value to `true`.

Reply:
I've added a commit to change the default values of `throw` and `failfast`.